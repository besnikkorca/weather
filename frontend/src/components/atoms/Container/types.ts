export enum BG {
  main = 'main',
}
export type Props = {
  children?: React.ReactNode;
  full?: boolean;
  bg: BG;
  maxWidth?: boolean;
  border?: boolean;
  centerContent?: boolean;
};
