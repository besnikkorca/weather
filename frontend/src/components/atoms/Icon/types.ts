import { CSSProperties } from 'react';
import { WeatherMain } from 'services/query/helpers/types';

export interface Props {
  name: WeatherMain;
  style?: CSSProperties | undefined;
  size: 'normal' | 'medium' | 'large';
  color?: 'main' | 'secondary';
}
