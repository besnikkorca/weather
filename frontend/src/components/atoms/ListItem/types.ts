export interface Props {
  children: React.ReactNode;
  active: boolean;
  onClick: (str: string) => void;
}
