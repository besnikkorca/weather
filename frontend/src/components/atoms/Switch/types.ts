export interface Props {
  checked: boolean;
  setChecked: (checked: boolean) => void;
}
