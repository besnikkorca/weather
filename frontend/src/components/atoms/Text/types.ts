import React from 'react';

export enum TextFamily {
  teko = 'teko',
}

export enum TextWeight {
  normal = 'normal',
  bold = 'bold',
}

export enum TextSize {
  small = 'small',
  normal = 'normal',
  large = 'large',
  xLarge = 'xLarge',
}

export enum TextColor {
  color1 = 'color1',
  color2 = 'color2',
  color3 = 'color3',
  color4 = 'color4',
}
export interface Props {
  noMargin?: boolean;
  children: React.ReactNode;
  size?: TextSize;
  weight?: TextWeight;
  family?: TextFamily | undefined;
  style?: React.CSSProperties | undefined;
  color?: TextColor;
}
