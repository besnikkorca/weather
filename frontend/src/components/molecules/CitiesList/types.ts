import { City } from 'templates/WeatherAppTemplate/types';

export interface Props {
  active: string;
  setActive: (active: string) => void;
  cities: City[];
}
