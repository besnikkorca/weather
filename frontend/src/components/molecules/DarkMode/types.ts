export interface Props {}

export type Theme = 'dark' | 'light';

export interface State {
  theme: Theme;
}
