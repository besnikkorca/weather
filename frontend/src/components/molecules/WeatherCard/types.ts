import { DayForecast } from 'src/components/services/query/helpers/types';

export interface Props {
  forecast: DayForecast | undefined;
}
