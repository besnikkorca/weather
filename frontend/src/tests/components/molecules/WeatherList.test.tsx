import { render } from '@testing-library/react';
import WeatherList from 'molecules/WeatherList';
import { DayForecast } from 'src/components/services/query/helpers/types';
import forecastList from 'src/tests/toydata/forecastList.json';

describe('WeatherList', () => {
  it('should render', () => {
    const { container } = render(<WeatherList forecasts={forecastList as DayForecast[]} />);

    expect(container).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const view = render(<WeatherList forecasts={forecastList as DayForecast[]} />);

    expect(view).toMatchSnapshot();
  });
});
