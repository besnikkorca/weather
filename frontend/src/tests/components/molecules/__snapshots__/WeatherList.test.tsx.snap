// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherList renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container transparentBG centerHorizontal container"
      >
        <div
          class="container transparentBG centerVertical border card firstCard"
        >
          <p
            class="default normal weightNormal color1"
          >
            Mon
          </p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cloud-rain col normal"
            data-icon="cloud-rain"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 320c-53 0-96-43-96-96c0-42.5 27.6-78.6 65.9-91.2C64.7 126.1 64 119.1 64 112C64 50.1 114.1 0 176 0c43.1 0 80.5 24.3 99.2 60c14.7-17.1 36.5-28 60.8-28c44.2 0 80 35.8 80 80c0 5.5-.6 10.8-1.6 16c.5 0 1.1 0 1.6 0c53 0 96 43 96 96s-43 96-96 96H96zm-6.8 52c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3L89.2 372zm160 0c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3L249.2 372zm124.9 64.6L409.2 372c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3z"
              fill="currentColor"
            />
          </svg>
          <p
            class="default noMargin small weightNormal teko color3"
          >
            3
            째
          </p>
        </div>
        <div
          class="container transparentBG centerVertical border card notFirstCard lastCard"
        >
          <p
            class="default normal weightNormal color1"
          >
            Mon
          </p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cloud-rain col normal"
            data-icon="cloud-rain"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 320c-53 0-96-43-96-96c0-42.5 27.6-78.6 65.9-91.2C64.7 126.1 64 119.1 64 112C64 50.1 114.1 0 176 0c43.1 0 80.5 24.3 99.2 60c14.7-17.1 36.5-28 60.8-28c44.2 0 80 35.8 80 80c0 5.5-.6 10.8-1.6 16c.5 0 1.1 0 1.6 0c53 0 96 43 96 96s-43 96-96 96H96zm-6.8 52c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3L89.2 372zm160 0c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3L249.2 372zm124.9 64.6L409.2 372c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3z"
              fill="currentColor"
            />
          </svg>
          <p
            class="default noMargin small weightNormal teko color3"
          >
            2
            째
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container transparentBG centerHorizontal container"
    >
      <div
        class="container transparentBG centerVertical border card firstCard"
      >
        <p
          class="default normal weightNormal color1"
        >
          Mon
        </p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-cloud-rain col normal"
          data-icon="cloud-rain"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 320c-53 0-96-43-96-96c0-42.5 27.6-78.6 65.9-91.2C64.7 126.1 64 119.1 64 112C64 50.1 114.1 0 176 0c43.1 0 80.5 24.3 99.2 60c14.7-17.1 36.5-28 60.8-28c44.2 0 80 35.8 80 80c0 5.5-.6 10.8-1.6 16c.5 0 1.1 0 1.6 0c53 0 96 43 96 96s-43 96-96 96H96zm-6.8 52c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3L89.2 372zm160 0c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3L249.2 372zm124.9 64.6L409.2 372c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3z"
            fill="currentColor"
          />
        </svg>
        <p
          class="default noMargin small weightNormal teko color3"
        >
          3
          째
        </p>
      </div>
      <div
        class="container transparentBG centerVertical border card notFirstCard lastCard"
      >
        <p
          class="default normal weightNormal color1"
        >
          Mon
        </p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-cloud-rain col normal"
          data-icon="cloud-rain"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 320c-53 0-96-43-96-96c0-42.5 27.6-78.6 65.9-91.2C64.7 126.1 64 119.1 64 112C64 50.1 114.1 0 176 0c43.1 0 80.5 24.3 99.2 60c14.7-17.1 36.5-28 60.8-28c44.2 0 80 35.8 80 80c0 5.5-.6 10.8-1.6 16c.5 0 1.1 0 1.6 0c53 0 96 43 96 96s-43 96-96 96H96zm-6.8 52c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3L89.2 372zm160 0c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3L249.2 372zm124.9 64.6L409.2 372c1.3-2.5 3.9-4 6.8-4s5.4 1.5 6.8 4l35.1 64.6c4.1 7.5 6.2 15.8 6.2 24.3v3c0 26.5-21.5 48-48 48s-48-21.5-48-48v-3c0-8.5 2.1-16.9 6.2-24.3z"
            fill="currentColor"
          />
        </svg>
        <p
          class="default noMargin small weightNormal teko color3"
        >
          2
          째
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
